### ISSUE TITLE
org.eclipse.jetty.util.thread.ShutdownThread should use an appropriate name to identify itself in Thread dump

### ISSUE THREAD

**Issue by gonphsn** [eclipse/jetty.project#4789]
9.4.27

java version "1.8.0_251"
Java(TM) SE Runtime Environment (build 1.8.0_251-b08)
Java HotSpot(TM) 64-Bit Server VM (build 25.251-b08, mixed mode)

Windows 10

The Thread:

`          Name="Thread-6"   Class="org.eclipse.jetty.util.thread.ShutdownThread"`

Running from:
```

Stacks

  |  org.eclipse.jetty.util.component.AbstractLifeCycle.start() AbstractLifeCycle.java:72                   |

  Generated by YourKit Java Profiler 2019.8-b137 April 17, 2020 01:22:13 PM
```

Shows that the ShutdownThread does not apply an appropriate name to itself in its constructor:

```
public class ShutdownThread extends Thread
{
    private static final Logger LOG = Log.getLogger(ShutdownThread.class);
    private static final ShutdownThread _thread = new ShutdownThread();
         
   ...

    /**
     * Default constructor for the singleton
     *
     * Registers the instance as shutdown hook with the Java Runtime
     */
    private ShutdownThread()
    {
    }
```

For purposes of identifying ownership of all Threads in a Thread dump, the ShutdownThread() should use:

    private ShutdownThread()
    {
      super("Jetty:ShutdownThread") 
    }

or whatever your naming convention is.

**Comment by lorban** [comment]
What do you exactly mean with _**running from**_ `AbstractLifeCycle.java:72`?

The `ShutdownThread` is usually registered as a shutdown hook when [setStopAtShutdown(true)](https://www.eclipse.org/jetty/javadoc/9.3.24.v20180605/org/eclipse/jetty/server/Server.html#setStopAtShutdown-boolean-) is called on the `Server` instance, and should only run when the JVM is shutting down.

While we could set this thread's name, I find it odd that you're seeing this thread while the JVM is alive. Or did I misunderstand and you actually are talking about seeing this thread during the JVM's shutdown?

**Comment by gonphsn** [comment]
The provided stack snippet was the top line of thread dump entry and the only mutually relevant context of where the the thread was running from, I omitted the rest. I don't recall I captured this at shutdown or not, but if you say this should only occur at shutdown, I'll defer to your judgement and agree that was probably the case. Thread should still have a name as best practice the next time there's a more urgent code change requirement in that area.
